File: 082_Remove_Duplicates_from_Sorted_List_II/082_1.java
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) return head;

        ListNode res = head;
        ListNode prev = null;
        while (res != null && res.next != null)
        {
            if (res.val == res.next.val)
            {
                int currVal = res.val;
                while (res != null && res.val == currVal)
                    res = res.next; 
                if (prev != null)
                    prev.next = res;
                else
                    head = res;
            }
            else
            {
                prev = res;
                res = res.next;
            }
        }
        return head;
    }
}

File: 082_Remove_Duplicates_from_Sorted_List_II/082_1.py
class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head is None:
            return head

        prev = None
        curr = head
        while curr and curr.next:
            if curr.val != curr.next.val:
                prev = curr
                curr = curr.next
            else:
                while curr.next and curr.val == curr.next.val:
                    curr = curr.next
                if prev is not None:
                    prev.next = curr.next
                else:
                    head = curr.next
                curr = curr.next
        return head